File: my_portpolio_web/update.js
const fs = require('fs');

const path = 'c:/Users/고석준/Desktop/my_portpolio_web/business.html';
let content = fs.readFileSync(path, 'utf8');

// Remove cursor and onclick from cards
content = content.replace(/ cursor: pointer;" onclick="showDetail\('[^']+'\)"/g, '"');

// Change margin of use-cases-section
content = content.replace('<section class="use-cases-section">', '<section class="use-cases-section" style="margin-top: 150px;">');

// Extract use-cases-section block
const useCasesStart = content.indexOf('<!-- 적용사례 섹션 -->');
const useCasesEnd = content.indexOf('</section>', useCasesStart) + '</section>'.length;
const useCasesBlock = content.substring(useCasesStart, useCasesEnd);

// Replace detail-section with just the useCasesBlock
const detailStart = content.indexOf('<!-- 클릭 시 나타나는 상세 설명 영역 -->');
const detailEnd = content.indexOf('</div>\r\n    <script>', detailStart); // Actually we can just find </section>\s*</div>
const regex = /<!-- 클릭 시 나타나는 상세 설명 영역 -->[\s\S]*?<\/section>\s*<\/div>/;
content = content.replace(regex, useCasesBlock);

// Remove scripts
content = content.replace(/<script>[\s\S]*?<\/script>/, '');

fs.writeFileSync(path, content, 'utf8');
console.log('Done JS');

File: update.py
import re

with open('c:/Users/고석준/Desktop/my_portpolio_web/business.html', 'r', encoding='utf-8') as f:
    content = f.read()

# Remove onclick and cursor:pointer from business-cards
content = re.sub(r'cursor:\s*pointer;[\"\']\s*onclick=[\"\']showDetail\([^\)]+\)', '\"', content)
# Wait, replacing `cursor: pointer;" onclick="showDetail('industrial')` with `"` 
content = content.replace(' cursor: pointer;" onclick="showDetail(\'industrial\')"', '"')
content = content.replace(' cursor: pointer;" onclick="showDetail(\'disaster\')"', '"')
content = content.replace(' cursor: pointer;" onclick="showDetail(\'service\')"', '"')

# Find the use-cases-section
use_cases_match = re.search(r'<!-- 적용사례 섹션 -->.*?<section class=\"use-cases-section\">(.*?)</section>', content, re.DOTALL)
if use_cases_match:
    use_cases = use_cases_match.group(0)
    # Add some margin-top to move it down
    use_cases = use_cases.replace('<section class="use-cases-section">', '<section class="use-cases-section" style="margin-top: 150px;">')
else:
    use_cases = ''

# Remove the detail-section
# detail-section starts at <!-- 클릭 시 나타나는 상세 설명 영역 -->
# and ends after </section> \n    </div>
content = re.sub(r'<!-- 클릭 시 나타나는 상세 설명 영역 -->.*?<div id="detail-section"[^>]*>.*?<!-- 적용사례 섹션 -->.*?</section>\s*</div>', use_cases, content, flags=re.DOTALL)

# Remove the scripts at the bottom
content = re.sub(r'<script>\s*function showDetail.*?function changeTimeline.*?</script>', '', content, flags=re.DOTALL)

with open('c:/Users/고석준/Desktop/my_portpolio_web/business.html', 'w', encoding='utf-8') as f:
    f.write(content)

print("Done")
